### DISCUSSION TITLE
Help upgrading from v6 to v7

### DISCUSSION THREAD

**Discussion by ahmedosama7450** [react-hook-form/react-hook-form#5041] [Q&A] [answered]
In v6, I used to assign `register` to the form ref but now in v7, I need to spread the return value of `register` into the input.
register doesn't only return `ref` but other props like `onChange` and `onBlue` and `name`.

My codebase has custom components where I expose the ref using `React.forwardRef` and then normally assign `register` to their ref. I also give them the name prop manually.

So, here is the question, Will that work the same ? Or do I also have to expose `onChange` and `onBlur` ? Why does v7 even need to assign them if you already have the `ref`

**Comment by bluebill1049** [comment, marked as answer]
As long as u pass down onChange, onBlur, name and ref, everything should be worked. We used to attach even listener at hook form level, now we leave it to react. It certainly has its own pros and cons, but overall V7 syntax should be react future proof.

**Reply by ahmedosama7450** [reply]
Okey, I thought you only need either `ref` or `onChange/onBlue`. So, I need to expose onChange and onBlur from my custom components now.
But It really looks like you take more than you need. You only need `ref` or `onChange/onBlue` for registration.

I don't really understand the pros of this. Is there a link where I can read more about that ?

**Reply by bluebill1049** [reply]
> Okey, I thought you only need either `ref` or `onChange/onBlue`. So, I need to expose onChange and onBlur from my custom components now.
> But It really looks like you take more than you need. You only need `ref` or `onChange/onBlue` for registration.
> 
> I don't really understand the pros of this. Is there a link where I can read more about that ?

U can go have a read on V7 RFC.
